src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeConverter.cs

```diff
@@ -76,32 +76,7 @@ public static InputEnumType CreateEnumType(ref Utf8JsonReader reader, string? id
                 throw new JsonException("Enum must have at least one value");
             }
 
-            InputPrimitiveType? currentType = null;
```

Comment:
Why remove the value type creation and validation? we may still need check the enum value types consistence.

Reply:
I will introduce a fix for this in a new PR to make it throws a proper exception